### PR TITLE
Add v1.28.0 API testdata

### PR DESCRIPTION
**Pull request by SataQiu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Add v1.28.0 API testdata
https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/api/testdata#populating-data-for-each-release

#### Which issue(s) this PR fixes:

xref #119958

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/etcd/etcd_storage_path_test.go [hunk 1]
```diff
@@ -189,7 +189,7 @@ func TestEtcdStoragePath(t *testing.T) {
 					previousNonAlphaVersions.Insert(version)
 				}
 			}
-			if len(previousNonAlphaVersions) > 0 && !previousNonAlphaVersions.Has(expectedGVK.Version) {
+			if len(previousNonAlphaVersions) > 0 && !strings.Contains(expectedGVK.Version, "alpha") && !previousNonAlphaVersions.Has(expectedGVK.Version) {
```
Comment (SataQiu) [inline test/integration/etcd/etcd_storage_path_test.go:192]: Fixes:
```sh
TestEtcdStoragePath/admissionregistration.k8s.io/v1beta1,_Resource=validatingadmissionpolicies
    etcd_storage_path_test.go:193: Previous releases understand non-alpha versions ["v1beta1"], but do not understand the expected current storage version "v1alpha1". This means a current server will store data in etcd that is not understood by a previous version.
```

**Comment by liggitt** [comment]
verified the content of the v1.28.0 folder matches the content of the HEAD folder at the v1.28.0 tag

/lgtm
/approve

File: test/integration/etcd/etcd_storage_path_test.go (hunk 1 above)
Comment (liggitt) [reply test/integration/etcd/etcd_storage_path_test.go:192]: opened https://github.com/kubernetes/kubernetes/pull/120018 to switch the storage version and improve the etcd test to catch this

Comment (jpbetz) [reply test/integration/etcd/etcd_storage_path_test.go:192]: This was an oversight. Thanks for opening the PR to fix.

**Comment by SataQiu** [comment]
rebased #120018
Please take a look @liggitt

**Pull request merged**